R/steps-rcmdcheck.R

```diff
@@ -68,31 +76,34 @@ RCMDcheck <- R6Class(
 #' via [remotes::install_deps()] with `dependencies = TRUE`,
 #' and updating all packages.
 #'
-#' This step uses a dedicated library,
-#' a subdirectory `tic-pkg` of the current user library
-#' (the first element of [.libPaths()]),
-#' for the checks.
-#' This is done to minimize conflicts between dependent packages
-#' and packages that are required for running the various steps.
-#'
-#' @param warnings_are_errors `[flag]`\cr
-#'   Should warnings be treated as errors? Default: `TRUE`.
-#' @param notes_are_errors `[flag]`\cr
-#'   Should notes be treated as errors? Default: `FALSE`.
 #' @param args `[character]`\cr
-#'   Passed to `[rcmdcheck::rcmdcheck()]`, default:
+#'   Passed to `rcmdcheck::rcmdcheck()`, default:
 #'   `c("--no-manual", "--as-cran")`.
 #' @param build_args `[character]`\cr
-#'   Passed to `[rcmdcheck::rcmdcheck()]`, default:
+#'   Passed to `rcmdcheck::rcmdcheck()`, default:
 #'   `"--force"`.
+#' @param error_on `[character]`\cr
+#'   Whether to throw an error on R CMD check failures. Note that the check is
+#'   always completed (unless a timeout happens), and the error is only thrown
+#'   after completion. If "never", then no errors are thrown. If "error", then
+#'   only ERROR failures generate errors. If "warning", then WARNING failures
+#'   generate errors as well. If "note", then any check failure generated an
+#'   error.
+#' @param repos `[character]`\cr
+#'   Passed to `rcmdcheck::rcmdcheck()`, default:
+#'   `getOption("repos")`.
+#' @param timeout `[numeric]`\cr
+#'   Passed to `rcmdcheck::rcmdcheck()`, default:
+#'   `Inf`.
 #' @export
-step_rcmdcheck <- function(warnings_are_errors = TRUE, notes_are_errors = FALSE,
-                           args = c("--no-manual", "--as-cran"),
-                           build_args = "--force") {
+step_rcmdcheck <- function(args = c("--no-manual", "--as-cran"),
+                           build_args = "--force", error_on = "warning",
+                           repos = getOption("repos"), timeout = Inf) {
```

Comment:
Should we add an ellipsis here, and check `...length()` (or `length(list(...))` in earlier R versions), to allow a smoother transition?

Reply:
Thanks. I don't think `...length()` was available in R 3.1 or 3.2, It's slightly faster than `length(list(...))` .

Reply:
Idk if there is a more elegant solution of accounting for different R versions than adding another if-statement (which seems unnecessary) so I just replaced the part with the "old" version.